pkg/cli/mirror/create.go

```diff
@@ -117,17 +125,33 @@ func (o MirrorOptions) Create(ctx context.Context, flags *pflag.FlagSet) error {
 		return err
 	}
 
-	// Run archiver
+	// Catch any cancel signals
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+	done := make(chan bool, 1)
+	go func() {
+		<-sigs
+		if err := rollbackMeta(); err != nil {
+			logrus.Error(err)
+		}
+		os.Exit(1)
```

Comment:
Does the signal notifier channel catch this? Probably not, but I’m not sure. I would simply let this goroutine run to completion, block on `done` or the function’s completion using `select`, and call `os.Exit(1)` if `done` is not empty.

Reply:
@estroz good advice. Going to back to the drawing board on this. I was really using the goroutine as a way to test the metadata rollback functionality without giving a lot of thought to the implementation of that itself. Since I still think that we should handle cancel signals I will go back and incorporate your suggestions and add some unit tests for this.